### ISSUE TITLE
ODE casual training

### ISSUE THREAD

**Issue by artem-riabinin** [thu-ml/Causal-Forcing#47]
Hello,

I have a question regarding the second-stage ODE training (stage 2). I trained a model on 5-second clips for 4000 steps using two different setups:

1) Standard ODE with bidirectional teacher;

2) Causal ODE as described in the paper with AR teacher (trained for 3000 steps, further steps showed no significant improvement).

In my experiments, the causal ODE produced more dynamic results, with sharper face preservation and less blurring compared to the bidirectional version. However, I noticed that the background degrades more significantly during the ODE stage with the causal approach.

https://github.com/user-attachments/assets/161b2d3e-9089-48ed-b058-6de40cd47c3e

Is this expected behavior?

Additionally, when generating videos after DMD training (stage 3) for 600 steps, I start to see noticeable background artifacts in comparison with DMD training after ODE with bidirectional teacher. 

https://github.com/user-attachments/assets/f53abe3c-c304-4444-8c17-016ce062534a

Could you please advise on what might be causing this issue and how it could be addressed?

Thank you in advance for your help!

**Comment by zhuhz22** [comment]
Could you provide the detailed settings so that we can better analyze the issue?

1. For these two stages, how much ODE paired data did you use respectively, and what were the batch size and learning rate? Is the task setting T2V, or some other task such as video editing? Also, what chunk_size is your model?

2. By “Standard ODE with a bidirectional teacher,” do you mean that the student is AR while the teacher, which generates the ODE paired data, is bidirectional? Or do you mean that both the student and the teacher are bidirectional?

3. What model did you use, and how many parameters does it have?

4. Most significantly, for the AR teacher, how many steps is it trained, and how does it perform?


BTW, in this ODE demo comparison, I think the one on the right is clearly better than the one on the left:

<img width="590" height="194" alt="Image" src="https://github.com/user-attachments/assets/c3f8dcc0-3e80-4462-a879-aa0a69e022aa" />

Also, could you specifically describe what you mean by the “noticeable background artifacts” after DMD in this demo? Frankly speaking I didn’t notice any significant difference in the background.

**Comment by artem-riabinin** [comment]
1. For stages 1 and 2, I used approximately 200k paired clips, the batch size was 64, and lr was 5e-6. For stage 3, the batch size was also 64 and lr was 4e-7. I am adapting the Wan S2V autoregressive architecture from the Live Avatar paper for avatar generation conditioned on a reference image and audio input. Chunk_size is 4 frame latents.
2. To clarify: by "standard ODE with a bidirectional teacher," I mean that the teacher is bidirectional and is used to generate ODE trajectory targets, while the student is AR.
3. I am using the S2V model with approximately 2B parameters. Additionally, in the final stage, I experimented with using both real and fake scores (rather than only real score) with a larger 14B+ parameter model, but so far this has not led to significant improvements.
4. AR was trained for 3000 steps on 5-second clips using the teacher forcing approach; further steps showed no significant improvement. Here is its performance.

https://github.com/user-attachments/assets/b761520d-104d-40bb-997a-3a2b4ad4c627

By "noticeable background artifacts," I mean a grid-like pattern that starts appearing on the video (left — DMD after bidirectional ODE, right — DMD after causal ODE).

<img width="815" height="271" alt="Image" src="https://github.com/user-attachments/assets/5ad694f2-c69c-4602-9ff6-451a54102686" />

**Comment by zhuhz22** [comment]
I think the core issue lies in the artifacts of the AR teacher. From the video, I can see severe background issues, which are then inherited by ODE distillation. You may consider finding ways to improve the AR teacher.

Also, for DMD, you may consider training for fewer steps.

**Comment by artem-riabinin** [comment]
Thanks, I'll give that a try
